### PR TITLE
Setup crank for performance tests

### LINKED ISSUE #2089: Configure crank to run performance tests
Configure performance tests to run with [Microsoft.Crank](https://github.com/dotnet/crank)

This will make perf tests easier to run locally and remotely

### Assemblies affected

*Which assemblies and versions are known to be affected e.g. OData .Net lib 7.x*

### Reproduce steps

*The simplest set of steps to reproduce the issue. If possible, reference a commit that demonstrates the issue.*

### Expected result

*What would happen if there wasn't a bug.*

### Actual result

*What is actually happening.*

### Additional detail

*Optional, details of the root cause if known. Delete this section if you have no additional details to add.*

### PR DESCRIPTION
**Pull request by habbes**
### Issues

Fixes #2089

### Description

Adds [Microsoft.Crank](https://github.com/dotnet/crank) configuration and instructions for running benchmark tests.

Instructions on how to setup crank and run the benchmarks have been added to the README.

This PR has a lot file changes, but most of the changes are accidental fixes to the whitespace formatting, the main file changes are:

- `benchmarks.yml`: this contains the `crank` configuration. It tells crank how to build our benchmark projects and which scenarios to run. I tried to make it easy to run different scenarios (component tests, e2e tests, running against the main repo, etc.) and added instructions to the README.md
- `README.md`: added instructions on how to setup and run the benchmarks
- `OData.Test.Performance.sln`, `Microsoft.OData.Performance.ComponentTests.csproj`, `Microsoft.OData.Performance.ServiceTests.csproj`: The `crank-agent` was getting a hard time trying to build the projects, it was not able to properly parse the `.csproj` config. To fix this I re-created the solution and projects, now they have simpler configuration and only support .Net Core 3.1 (It should be possible to add support for net47 I believe). The IDE formatted the whitespace in the source files during this process, that's why there are many changed files. In the process I also removed obsolete `app.config`, `packages.config` and `AssemblyInfo.cs` files.
- `TestServiceFixture.cs`: this class is used by the "e2e" service tests to launch the test web service. It was using an env variable to locate the path of the test service. This was not easy to pass correctly to the crank config, so I updated the code so that it can manually find the directory to the test service.
- `PerformanceBuild.ps1`: this is the old powershell script for building and running the performance benchmarks. I updated it to work with the changed project files. You can use it to run perf tests without having to setup crank.

### Checklist (Uncheck if it is not completed)

- [ ] *Test cases added*
- [ ] *Build and test with one-click build and test script passed*

### Additional work necessary

*If documentation update is needed, please add "Docs Needed" label to the issue and provide details about the required document change in the issue.*

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -115,6 +115,92 @@ You can query the latest nightly NuGet packages using this query: [MAGIC OData q
...
+    crank --config benchmarks.yml --scenario UriParser --profile local
+    ```
+
+#### Run benchmarks against the official repo
+
+To run benchmarks against the official repo instead of your local repo, pass
+the `base=true` variable to the command, e.g.:
+
+```text
+crank --config benchmarks.yml --scenario Service --profile local --variable base=true
```
Comment (marabooy) [inline README.md:193]: Should this be local?

Comment (marabooy) [reply README.md:193]: I see if base=true then it will use the remote. To avoid confusion maybe call this out explicitly.

**Review by marabooy — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (habbes) [reply README.md:193]: It's local in the sense it runs locally on your machine. In the next PRs I intend to setup remote agents so you can run benchmarks on remote servers. But you will still have the option to either run your local copy of the code or the official repo. So `base` is probably not a good name, but `local` would be confusing. `base` means the official repo will be cloned, so any changes you've made locally won't be picked. It's meant to allow you to compare how changes affect the benchmarks compared to the official repo.

File: README.md [hunk 2]
```diff
@@ -115,6 +115,92 @@ You can query the latest nightly NuGet packages using this query: [MAGIC OData q
 
 The release of the component binaries is carried out regularly through [Nuget](http://www.nuget.org/).
 
+### 3.6 Performance benchmarks
+
+#### Installation
+
+The easiest way to run the perf benchmarks is to use the [Microsoft.Crank](https://github.com/dotnet/crank) toolset.
+
+- Install the [Crank controller](https://www.nuget.org/packages/Microsoft.Crank.Controller), the CLI use to run benchmarks:
```
Comment (gathogojr) [inline README.md:124]:
```suggestion
- Install the [Crank controller](https://www.nuget.org/packages/Microsoft.Crank.Controller), the CLI used to run benchmarks:
```

File: README.md [hunk 3]
```diff
@@ -115,6 +115,92 @@ You can query the latest nightly NuGet packages using this query: [MAGIC OData q
...
+#### Installation
+
+The easiest way to run the perf benchmarks is to use the [Microsoft.Crank](https://github.com/dotnet/crank) toolset.
+
+- Install the [Crank controller](https://www.nuget.org/packages/Microsoft.Crank.Controller), the CLI use to run benchmarks:
+
+    ```text
+    dotnet tool install -g --version "0.2.0-*"
+    ```
+- Install the [Crank agent](https://www.nuget.org/packages/Microsoft.Crank.Agent), service that executes benchmark jobs. This should be install on the server(s) where the benchmarks will run. Install locally if you intend to run benchmarks locally.
```
Comment (gathogojr) [inline README.md:129]:
```suggestion
- Install the [Crank agent](https://www.nuget.org/packages/Microsoft.Crank.Agent), service that executes benchmark jobs. This should be installed on the server(s) where the benchmarks will run. Install locally if you intend to run benchmarks locally.

```

File: benchmarks.yml [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+      job: components
+      variables:
+        filter: "*"
+  Service:
+    application:
+      job: service
+      variables:
+        filter: "*"
+
+
```
Comment (gathogojr) [inline benchmarks.yml:77]:
```suggestion
```

File: benchmarks.yml [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+profiles:
+  local:
+    jobs:
+      application:
+        endpoints:
+          - http://localhost:5010
```
Comment (gathogojr) [inline benchmarks.yml:83]:
```suggestion
          - http://localhost:5010

```

File: test/PerformanceTests/ComponentTests/Microsoft.OData.Performance.ComponentTests.csproj [hunk 6]
```diff
@@ -1,369 +1,40 @@
...
-    <Compile Include="Scenario\BatchRoundTripPerfTest.cs" />
-    <Compile Include="Scenario\WriteReadFeedTestBase.cs" />
-    <Compile Include="UriParser\UriParserTests.cs" />
-    <Compile Include="Program.cs" />
-  </ItemGroup>
-  <ItemGroup>
-    <EmbeddedResource Include="Common\Models\AdventureWorksPlus.csdl">
-      <SubType>Designer</SubType>
-    </EmbeddedResource>
+    <EmbeddedResource Include="Common\Models\AdventureWorksPlus.csdl" />
```
Comment (gathogojr) [inline test/PerformanceTests/ComponentTests/Microsoft.OData.Performance.ComponentTests.csproj:19]: _Common/Models/AdventureWorksPlus.dll_ appears [here](29696ef03dad61c01ac0642cc0a54d35fda5d833c2ef3d87cdfbf3af752c3848R9) with a **Compile Action** of **None** yet it also appears here as **EmbeddedResource**. Is it deliberate or inadvertent?

File: test/PerformanceTests/ServiceTests/AssemblyInfo.cs [hunk 7]
```diff
@@ -1,3 +0,0 @@
-﻿using Xunit;
-
-[assembly: CollectionBehavior(DisableTestParallelization = true)]
```
Comment (gathogojr) [inline test/PerformanceTests/ServiceTests/AssemblyInfo.cs:3]: I assume we no longer have any tests in this project using Xunit, no?

File: test/PerformanceTests/ServiceTests/UpdateTests.cs [hunk 8]
```diff
@@ -177,6 +177,6 @@ public void PatchEntity()
             }
         }
 
-        
+
```
Comment (gathogojr) [inline test/PerformanceTests/ServiceTests/UpdateTests.cs:180]: You could actually delete this line and the other one above it

**Review by gathogojr — APPROVED** [review]
Left a few minor comments otherwise looks great. I'll try it out locally

File: test/PerformanceTests/ServiceTests/AssemblyInfo.cs (hunk 7 above)
Comment (habbes) [reply test/PerformanceTests/ServiceTests/AssemblyInfo.cs:3]: We don't. Still using xunit for some assertions in the ServiceTests, but not for running the tests.

File: test/PerformanceTests/ComponentTests/Microsoft.OData.Performance.ComponentTests.csproj (hunk 6 above)
Comment (habbes) [reply test/PerformanceTests/ComponentTests/Microsoft.OData.Performance.ComponentTests.csproj:19]: On VS it appeared as EmbeddedResource. Clicking the link you shared leads to a 404 so I'm not sure which file you're referencing.

The tests except for the files to be embedded resources cause that's where they try to read them from.

Tests call the `TestUtils.ReadTestResource` method which aemots to load the file from embedded resource.

**Pull request merged**